### PR TITLE
added http copy test

### PR DESCRIPTION
**Pull request by AndoniSanguesa**
Signed-Off By: Andoni Sanguesa

### Description of PR

Summary:
Adds a test that checks that the DUT is able to access external objects via HTTP.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
This test existed in the Arista testplan but was not present for SONiC.

#### How did you do it?
An HTTP server is started on the localhost where a large file is downloaded. The DUT uses `curl` to fetch the file from the HTTP server. File integrity is checked by comparing checksums for the file before and after transport.

#### How did you verify/test it?
The test passed on my platform and confirmed that the file was copied to the host and the dut manually. 

#### Any platform specific information?
I've been encountering the issue found in #3778 , so I've had to skip the sanity checks to confirm that this test passes. This shouldn't be an issue since this test does not interact with the T1 leaf nodes in any way. The only communication necessary is between the host and the DUT.

#### Supported testbed topology if it's a new test case?
Tested on t0.

 But should work on any topology since this test is only using direct communication between the host and the DUT.

### Documentation 

Wrote short [testplan](https://github.com/AndoniSanguesa/sonic-mgmt/blob/test_http_copy/docs/testplan/HTTP_copy-test-plan.md) for test.

### CODE REVIEW

**Comment by yxieca** [comment]
@AndoniSanguesa I suggest refactoring this test:

1. Run the HTTP server on the PTF docker, so that you don't have to hard code the HTTP server IP. PTF IP is available in configuration. You can reference the tacacs tests.
2. You can generate a file from /dev/random with a controlled size, so that avoid downloading from a website that might be not available at the time of the test.

**Comment by AndoniSanguesa** [comment]
Thanks for the comments @yxieca. I just made the changes.

File: tests/http/test_http_copy.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    started = False
+    tries = 0
+    while not started and tries < 10:
+        if os.system("curl {}:8080".format(ptf_ip)) == 0:
+            started = True
+        tries += 1
+        time.sleep(1)
+
+    if not started:
+        pytest.fail("HTTP Server could not be started")
```
Comment (yxieca) [inline tests/http/test_http_copy.py:40]: We prefer use the pytest_assert to make the code compact. Again, you can reference tacacs test for example:

https://github.com/Azure/sonic-mgmt/blob/master/tests/tacacs/test_ro_user.py#L3

https://github.com/Azure/sonic-mgmt/blob/master/tests/tacacs/test_ro_user.py#L127

Sorry should have raised this with the other comments. Please address other places in this file below as well.

Comment (AndoniSanguesa) [reply tests/http/test_http_copy.py]: Great, thanks again for the suggestion @yxieca, I hadn't seen that in the tests I was using for reference. I've just pushed the change.

**Review by yxieca — APPROVED** [review]

File: tests/http/test_http_copy.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    pytest_assert(res!=0, "DUT container could not be cleaned.")
+
+    # Delete file off ptf
+    ptfhost.command(("rm ./{}".format(test_file_name)))
+
+    # Ensure that file was removed correctly
+    res = ptfhost.command("ls -ltr ./{}".format(test_file_name), module_ignore_errors=True)["rc"]
+
+    pytest_assert(res!=0, "PTF container could not be cleaned.")
```
Comment (yxieca) [inline tests/http/test_http_copy.py:82]: Should this condition be res == 0?

Comment (yxieca) [reply tests/http/test_http_copy.py:82]: oh, never mind.

**Comment by yxieca** [comment]
@AndoniSanguesa this test has been consistently causing ptf docker to crash after the test is done and causing nightly test to fail afterwards. Please investigate the issue and propose the change with a new PR.

**Pull request merged**